plugins/catalog-backend/src/catalog/types.ts

```diff
@@ -251,29 +251,27 @@ export type Cursor = {
    */
   sortFields: EntitySortField[];
   /**
-   * The value of the cursor of the last item returned.
-   * This is used for performing pagination.
+   * The values of the fields of a specific item used for paginating the data.
    */
-  sortFieldIds: string[];
-
+  sortFieldValues: string[];
```

Comment:
Noting here that `null` is an actual valid value in the search table 🤔 It's used when the value is actual null, or when the value is so large that it didn't fit in the value column. 

https://github.com/backstage/backstage/blob/8ba579e8bd47793492e3e8f0b8c8077399a7bb45/plugins/catalog-backend/src/stitching/buildEntitySearch.ts#L82

https://github.com/backstage/backstage/blob/8ba579e8bd47793492e3e8f0b8c8077399a7bb45/plugins/catalog-backend/src/stitching/buildEntitySearch.ts#L146

Reply:
It's perfectly possible that the sort field values aren't enough to distinguish where to start the next page, right?

Like, if you have `sortFields: ['kind']`, it's very likely that the value is the same across pages. So if you want stable sort, do you have to still put for example `metadata.uid` as the last key? That's not perfect either i guess 🤔 Hm this stuff is tricky

Reply:
> Noting here that null is an actual valid value in the search table 

oh I didn't know! I will update the typings.

> So if you want stable sort, do you have to still put for example metadata.uid as the last key?

yes, this is exactly what this PR does! The `sortFieldValues` array will contain the value of the sort field choosed at request time at position 0 together with `entity_id` at position 1. It's like if the request contained `sortFields: ['kind', 'metadata.uid']` excepting that `metadata.uid` is kind of hardcoded at the moment, which is fine since we don't support multiple sortFields at the moment.

https://github.com/backstage/backstage/pull/15513/files#diff-7841edf78cd947efd000b8b63a59d6bfbe1bb55c20e76fb810b2fedb33211aa2R286

It looks like variable naming needs to be improved 😅 I will update!

Reply:
Addressed the two issues in:

https://github.com/backstage/backstage/pull/15513/commits/5c4d8cf94efd1cd8e1a37481629724bb02079113
https://github.com/backstage/backstage/pull/15513/commits/3c6d0953ee5ce842636fe93b79146a59a722004b